File: introductionToAlgorithmCLRS/chp2/Exercise/Ex_2.2/q2SelectionSort.java
import java.util.Arrays;
import java.util.Scanner;


public class q2SelectionSort {

    public static int[] selectionSort(int array[], int size) {
        for (int i = 0 ; i < size; i++) {
            int smallest = array[i];
            for (int j = i; j < size; j++) {
                if (smallest > array[j]) {
                    int temp = array[i];
                    array[i] = array[j];
                    array[j] = temp;
                    smallest = array[j];
                }
            }
        }
        return array;
    }
    public static void main(String[] args) {
        System.out.print("Enter the size of array : ");
        Scanner element = new Scanner(System.in);
        int size  = element.nextInt();
        int array[] = new int[size];
        System.out.print("Enter elements of the array : ");
        for (int i = 0 ; i < size; i++ ) {
            array[i] = element.nextInt();
        }
        element.close();
        
        int sortedArray[] = new int[size];
        sortedArray = selectionSort(array, size);
        System.out.println("Sorted Array is : " + Arrays.toString(sortedArray));

    }
}

File: introductionToAlgorithmCLRS/chp2/Exercise/Ex_2.2/q2SelectionSort.py
def selectionSort(array, size):
    for i in range(size):
        smallest = array[i]
        for j in range(i, size):
            if (smallest > array[j]):
                temp = array[i]
                array[i] = array[j]
                array[j] = temp
                smallest = array[j]
    return array


size = int(input("Enter the size of array : "))
array = list(map(lambda x: int(x), input("Enter elements of array : ").split()))


sortedArray = selectionSort(array, size)
print("Sorted Array is : ", sortedArray)

